<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Estimate log fold changes using DESeq2. — deseqfun • power.nb</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Estimate log fold changes using DESeq2. — deseqfun"><meta name="description" content="This function estimates log fold changes (LFC) for microbiome count data using the DESeq2 package.
It is strongly recommended to keep the following defaults:
minReplicatesForReplace=Inf
cooksCutoff=TRUE
independentFiltering=TRUE
These options are particularly useful when estimating fold changes to fit the mixture of Gaussian distributions.

"><meta property="og:description" content="This function estimates log fold changes (LFC) for microbiome count data using the DESeq2 package.
It is strongly recommended to keep the following defaults:
minReplicatesForReplace=Inf
cooksCutoff=TRUE
independentFiltering=TRUE
These options are particularly useful when estimating fold changes to fit the mixture of Gaussian distributions.

"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">power.nb</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/stub.html">Power and Sample Size Estimation for Microbiome Analysis</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Estimate log fold changes using <code>DESeq2</code>.</h1>

      <div class="d-none name"><code>deseqfun.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function estimates log fold changes (LFC) for microbiome count data using the DESeq2 package.
It is strongly recommended to keep the following defaults:</p><ul><li><p><code>minReplicatesForReplace=Inf</code></p></li>
<li><p><code>cooksCutoff=TRUE</code></p></li>
<li><p><code>independentFiltering=TRUE</code>
These options are particularly useful when estimating fold changes to fit the mixture of Gaussian distributions.</p></li>
</ul></div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">deseqfun</span><span class="op">(</span></span>
<span>  <span class="va">countdata</span>,</span>
<span>  <span class="va">metadata</span>,</span>
<span>  alpha_level <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  ref_name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">group_colname</span>,</span>
<span>  <span class="va">sample_colname</span>,</span>
<span>  minReplicatesForReplace <span class="op">=</span> <span class="cn">Inf</span>,</span>
<span>  cooksCutoff <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  independentFiltering <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  shrinkage_method <span class="op">=</span> <span class="st">"normal"</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-countdata">countdata<a class="anchor" aria-label="anchor" href="#arg-countdata"></a></dt>
<dd><p>A matrix of OTU count data where rows represent taxa and columns represent samples.</p></dd>


<dt id="arg-metadata">metadata<a class="anchor" aria-label="anchor" href="#arg-metadata"></a></dt>
<dd><p>A dataframe containing sample information with two rows: one for sample names and one for group names.</p></dd>


<dt id="arg-alpha-level">alpha_level<a class="anchor" aria-label="anchor" href="#arg-alpha-level"></a></dt>
<dd><p>The significance level for determining differential expression. Default is 0.1.</p></dd>


<dt id="arg-ref-name">ref_name<a class="anchor" aria-label="anchor" href="#arg-ref-name"></a></dt>
<dd><p>reference level for fold change calculation.
If NULL, the reference level is determined automatically — by default,
the factor level that comes first is used as the reference.</p></dd>


<dt id="arg-group-colname">group_colname<a class="anchor" aria-label="anchor" href="#arg-group-colname"></a></dt>
<dd><p>column names of the groups or conditions</p></dd>


<dt id="arg-sample-colname">sample_colname<a class="anchor" aria-label="anchor" href="#arg-sample-colname"></a></dt>
<dd><p>column names of the samples</p></dd>


<dt id="arg-minreplicatesforreplace">minReplicatesForReplace<a class="anchor" aria-label="anchor" href="#arg-minreplicatesforreplace"></a></dt>
<dd><p>DESeq2's parameter to control the minimum number of replicates required for replacing outliers during dispersion estimation. Default is <code>Inf</code> (no replacement).</p></dd>


<dt id="arg-cookscutoff">cooksCutoff<a class="anchor" aria-label="anchor" href="#arg-cookscutoff"></a></dt>
<dd><p>DESeq2's parameter for removing outliers based on Cook's distance. Default is <code>TRUE</code> (outlier removal enabled).</p></dd>


<dt id="arg-independentfiltering">independentFiltering<a class="anchor" aria-label="anchor" href="#arg-independentfiltering"></a></dt>
<dd><p>DESeq2's parameter for independent filtering. Default is <code>TRUE</code>.</p></dd>


<dt id="arg-shrinkage-method">shrinkage_method<a class="anchor" aria-label="anchor" href="#arg-shrinkage-method"></a></dt>
<dd><p>DESeq2's shrinkage method for fold changes. Default is <code>"normal"</code>. Other options include <code>"apeglm"</code> or <code>"ashr"</code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A list containing the following elements:</p><ul><li><p><code>logfoldchange</code>: A vector of log fold change estimates for each taxa.</p></li>
<li><p><code>dispersion</code>: A vector of dispersion estimates for each taxa.</p></li>
<li><p><code>deseq_estimate</code>: A dataframe containing DESeq2 results, including <code>baseMean</code>, <code>log2FoldChange</code>, <code>lfcSE</code>, <code>pvalue</code>, and <code>padj</code>.</p></li>
<li><p><code>normalised_count</code>: A matrix of normalized count data.</p></li>
<li><p><code>dds</code>: DESeq object</p></li>
</ul></div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># Example usage</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">101</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">nr</span> <span class="op">=</span> <span class="fl">10</span>; <span class="va">nc</span> <span class="op">=</span><span class="fl">50</span></span></span>
<span class="r-in"><span><span class="va">countdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Poisson.html" class="external-link">rpois</a></span><span class="op">(</span><span class="fl">500</span>, <span class="fl">3</span><span class="op">)</span>, ncol <span class="op">=</span> <span class="va">nc</span>, nrow <span class="op">=</span> <span class="va">nr</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># Simulated OTU count data with 50 taxa and 10 samples</span></span></span>
<span class="r-in"><span><span class="va">countdata</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">countdata</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">countdata</span><span class="op">)</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Sample"</span>, <span class="fl">1</span><span class="op">:</span><span class="va">nr</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">countdata</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"otu"</span>, <span class="fl">1</span><span class="op">:</span><span class="va">nc</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>Samples <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Sample"</span>, <span class="fl">1</span><span class="op">:</span><span class="va">nr</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>             Groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Control"</span>, <span class="st">"Treatment"</span><span class="op">)</span>, each <span class="op">=</span> <span class="fl">5</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">sample_colname</span> <span class="op">=</span> <span class="st">"Samples"</span></span></span>
<span class="r-in"><span><span class="va">group_colname</span>  <span class="op">=</span> <span class="st">"Groups"</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu">deseqfun</span><span class="op">(</span><span class="va">countdata</span>, <span class="va">metadata</span>, ref_name <span class="op">=</span> <span class="st">"Control"</span>,</span></span>
<span class="r-in"><span>                    minReplicatesForReplace <span class="op">=</span> <span class="cn">Inf</span>,</span></span>
<span class="r-in"><span>                    cooksCutoff <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>                    sample_colname <span class="op">=</span> <span class="st">"Samples"</span>,</span></span>
<span class="r-in"><span>                    group_colname  <span class="op">=</span> <span class="st">"Groups"</span>,</span></span>
<span class="r-in"><span>                    independentFiltering <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>                    shrinkage_method<span class="op">=</span><span class="st">"normal"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> estimating size factors</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> estimating dispersions</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> gene-wise dispersion estimates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> mean-dispersion relationship</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -- note: fitType='parametric', but the dispersion trend was not well captured by the</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    function: y = a/x + b, and a local regression fit was automatically substituted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    specify fitType='local' or 'mean' to avoid this message next time.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> final dispersion estimates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> fitting model and testing</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> using 'normal' for LFC shrinkage, the Normal prior from Love et al (2014).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Note that type='apeglm' and type='ashr' have shown to have less bias than type='normal'.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> See ?lfcShrink for more details on shrinkage type, and the DESeq2 vignette.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reference: https://doi.org/10.1093/bioinformatics/bty895</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Examine the results</span></span></span>
<span class="r-in"><span><span class="va">result</span><span class="op">$</span><span class="va">logfoldchange</span>  <span class="co"># Log fold changes</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        otu_1        otu_2        otu_3        otu_4        otu_5        otu_6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> -0.007959025 -0.445880472 -0.394760811  0.233294799 -0.091829484  0.058651776 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        otu_7        otu_8        otu_9       otu_10       otu_11       otu_12 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> -0.308246138 -0.308896891 -0.251029922 -0.252289726  0.039529520  0.160924136 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       otu_13       otu_14       otu_15       otu_16       otu_17       otu_18 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  0.194703220  0.192716051 -0.464400160  0.068904985 -0.097712013  0.053908758 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       otu_19       otu_20       otu_21       otu_22       otu_23       otu_24 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> -0.363959631  0.156101373 -0.144220223 -0.013117419  0.028005740  0.150500957 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       otu_25       otu_26       otu_27       otu_28       otu_29       otu_30 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  0.042978792  0.027651694  0.141706746 -0.177098228  0.456759743 -0.309509409 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       otu_31       otu_32       otu_33       otu_34       otu_35       otu_36 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  0.035671304 -0.228363278  0.030497385 -0.256186889  0.044228160 -0.083204924 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       otu_37       otu_38       otu_39       otu_40       otu_41       otu_42 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  0.251044749  0.271340396 -0.148152551 -0.417094381  0.003548958  0.101448484 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       otu_43       otu_44       otu_45       otu_46       otu_47       otu_48 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> -0.023449549 -0.062952952 -0.208034179 -0.104467686 -0.019733425  0.030671416 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       otu_49       otu_50 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  0.273876722  0.149311832 </span>
<span class="r-in"><span><span class="va">result</span><span class="op">$</span><span class="va">logmean</span>  <span class="co"># Log mean counts</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> NULL</span>
<span class="r-in"><span><span class="va">result</span><span class="op">$</span><span class="va">dispersion</span>  <span class="co"># Dispersion estimates</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] 0.23726334 0.11818891 0.12779467 0.20192426 0.17595129 0.07919315</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [7] 0.11134903 0.14821130 0.42777852 0.19140022 0.85818539 0.12657805</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [13] 0.24094148 0.07205231 0.09815225 0.07600581 0.14666306 0.04598340</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [19] 0.25463325 0.16538272 0.08420186 0.23473744 0.06856281 0.17004895</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [25] 0.22618425 0.06933439 0.07672390 0.12162285 0.14324368 0.14440827</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [31] 0.23368195 0.08659100 0.11722574 0.24416527 0.13947440 0.28604738</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [37] 0.08057311 0.11586475 0.07294106 0.07079641 0.13274784 0.07402237</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [43] 0.14003320 0.11678770 0.13505417 0.06700914 0.06674076 0.10383276</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [49] 0.21379294 0.17696335</span>
<span class="r-in"><span><span class="va">result</span><span class="op">$</span><span class="va">deseq_estimate</span>  <span class="co"># DESeq2 results</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        baseMean log2FoldChange     lfcSE        stat     pvalue      padj</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_1  2.523916   -0.007959025 0.2519496 -0.03164913 0.97475186 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_2  3.436181   -0.445880472 0.2632859 -1.67359558 0.09421011 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_3  3.542517   -0.394760811 0.2633018 -1.48732907 0.13692792 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_4  2.702925    0.233294799 0.2553894  0.91069198 0.36245769 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_5  2.694906   -0.091829484 0.2572253 -0.35677492 0.72126031 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_6  3.287608    0.058651776 0.2642521  0.22185078 0.82443004 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_7  4.150850   -0.308246138 0.2647080 -1.16052986 0.24583314 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_8  3.555650   -0.308896891 0.2626516 -1.17228830 0.24108133 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_9  3.090175   -0.251029922 0.2446089 -1.02344368 0.30609811 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_10 2.675081   -0.252289726 0.2561637 -0.97892159 0.32761872 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_11 2.749900    0.039529520 0.2178270  0.18227420 0.85536754 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_12 3.390569    0.160924136 0.2630611  0.61090560 0.54126208 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_13 2.356214    0.194703220 0.2500851  0.77559202 0.43798991 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_14 3.205661    0.192716051 0.2641421  0.72725190 0.46707166 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_15 3.301207   -0.464400160 0.2633714 -1.74236099 0.08144530 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_16 3.130311    0.068904985 0.2638552  0.26125914 0.79389267 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_17 4.059487   -0.097712013 0.2640602 -0.36948165 0.71176875 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_18 4.936842    0.053908758 0.2616827  0.20581280 0.83693715 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_19 2.433494   -0.363959631 0.2494307 -1.44767679 0.14770746 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_20 2.736376    0.156101373 0.2577149  0.60598515 0.54452462 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_21 3.243073   -0.144220223 0.2639479 -0.54652933 0.58470213 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_22 2.633467   -0.013117419 0.2535946 -0.05154981 0.95888741 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_23 3.018337    0.028005740 0.2637688  0.10623129 0.91539885 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_24 2.658367    0.150500957 0.2572649  0.58326522 0.55971477 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_25 2.140747    0.042978792 0.2488796  0.17205591 0.86339357 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_26 3.019768    0.027651694 0.2637470  0.10490246 0.91645321 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_27 3.255246    0.141706746 0.2642600  0.53474221 0.59282813 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_28 3.335342   -0.177098228 0.2630580 -0.67289343 0.50101509 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_29 1.956007    0.456759743 0.2498016  1.76909120 0.07687866 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_30 3.517628   -0.309509409 0.2627584 -1.17253684 0.24098159 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_31 2.388890    0.035671304 0.2509830  0.14209553 0.88700455 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_32 3.218515   -0.228363278 0.2638465 -0.86312957 0.38806623 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_33 2.797655    0.030497385 0.2609264  0.11688845 0.90694845 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_34 2.127118   -0.256186889 0.2466555 -1.03492011 0.30070622 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_35 3.643876    0.044228160 0.2634726  0.16759310 0.86690340 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_36 2.505223   -0.083204924 0.2488969 -0.33428091 0.73816759 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_37 2.873749    0.251044749 0.2627610  0.94886216 0.34269072 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_38 3.371951    0.271340396 0.2635050  1.02123215 0.30714446 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_39 3.037822   -0.148152551 0.2636697 -0.56179228 0.57425755 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_40 3.034731   -0.417094381 0.2635909 -1.56179566 0.11833613 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_41 3.759194    0.003548958 0.2639096  0.01343493 0.98928080 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_42 3.121752    0.101448484 0.2641010  0.38304575 0.70168584 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_43 3.885608   -0.023449549 0.2639198 -0.08883389 0.92921392 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_44 3.571744   -0.062952952 0.2639640 -0.23829804 0.81164994 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_45 3.560160   -0.208034179 0.2633832 -0.78692916 0.43132333 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_46 2.933184   -0.104467686 0.2634991 -0.39666457 0.69161483 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_47 3.075949   -0.019733425 0.2640999 -0.07469519 0.94045724 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_48 2.803747    0.030671416 0.2615479  0.11730384 0.90661928 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_49 2.045793    0.273876722 0.2476936  1.09437629 0.27378998 0.9892808</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_50 2.686524    0.149311832 0.2569471  0.58024839 0.56174712 0.9892808</span>
<span class="r-in"><span><span class="va">result</span><span class="op">$</span><span class="va">normalised_count</span>  <span class="co"># Normalized count data</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> NULL</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Michael Agronah, Ben Bolker.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer></div>





  </body></html>

